packages/dbml-core/src/export/MysqlExporter.js

```diff
@@ -134,29 +188,58 @@ class MySQLExporter {
       const foreignEndpointSchema = model.schemas[foreignEndpointTable.schemaId];
       const foreignEndpointFieldName = this.buildFieldName(foreignEndpoint.fieldIds, model, 'mysql');
 
-      let line = `ALTER TABLE ${shouldPrintSchema(foreignEndpointSchema, model)
-        ? `\`${foreignEndpointSchema.name}\`.` : ''}\`${foreignEndpointTable.name}\` ADD `;
+      if (refOneIndex === -1) {
+        const keyTableFirst = this.buildFieldKeyTableFirst(refEndpoint.fieldIds, model, 'mysql');
+        const keyTableSecond = this.buildFieldKeyTableSecond(foreignEndpoint.fieldIds, model, keyTableFirst);
 
-      if (ref.name) {
-        line += `CONSTRAINT \`${ref.name}\` `;
-      }
+        const nameNewTable = this.buildNameNewTable(refEndpointTable.name, foreignEndpointTable.name, model.tables);
+        line += this.buildTableManyToMany(keyTableFirst, keyTableSecond, nameNewTable);
 
-      line += `FOREIGN KEY ${foreignEndpointFieldName} REFERENCES ${shouldPrintSchema(refEndpointSchema, model)
-        ? `\`${refEndpointSchema.name}\`.` : ''}\`${refEndpointTable.name}\` ${refEndpointFieldName}`;
-      if (ref.onDelete) {
-        line += ` ON DELETE ${ref.onDelete.toUpperCase()}`;
-      }
-      if (ref.onUpdate) {
-        line += ` ON UPDATE ${ref.onUpdate.toUpperCase()}`;
+        if (keyTableFirst.size > 1) {
+          line += this.buildIndexManytoMany(keyTableFirst, nameNewTable, refEndpointTable.name);
+        }
+
+        if (keyTableSecond.size > 1) {
+          line += this.buildIndexManytoMany(keyTableSecond, nameNewTable, foreignEndpointTable.name);
+        }
+        line += this.buildForeignKeyManyToMany(keyTableFirst, refEndpointFieldName, nameNewTable, refEndpointTable.name, refEndpointSchema, model);
+        line += this.buildForeignKeyManyToMany(keyTableSecond, foreignEndpointFieldName, nameNewTable, foreignEndpointTable.name, foreignEndpointSchema, model);
+      } else {
+        line = `ALTER TABLE ${shouldPrintSchema(foreignEndpointSchema, model)
+          ? `\`${foreignEndpointSchema.name}\`.` : ''}\`${foreignEndpointTable.name}\` ADD `;
+        if (ref.name) {
+          line += `CONSTRAINT \`${ref.name}\` `;
+        }
+        line += `FOREIGN KEY ${foreignEndpointFieldName} REFERENCES ${shouldPrintSchema(refEndpointSchema, model)
+          ? `\`${refEndpointSchema.name}\`.` : ''}\`${refEndpointTable.name}\` ${refEndpointFieldName}`;
+        if (ref.onDelete) {
+          line += ` ON DELETE ${ref.onDelete.toUpperCase()}`;
+        }
+        if (ref.onUpdate) {
+          line += ` ON UPDATE ${ref.onUpdate.toUpperCase()}`;
+        }
+        line += ';\n';
       }
-      line += ';\n';
 
       return line;
     });
 
     return strArr;
   }
 
+  static buildNameNewTable (tableFirst, tableSecond, tables) {
+    let nameNewTable = `${tableFirst}_${tableSecond}`;
+    let count = 1;
+    while (true) {
```

Comment:
This is very inefficient and won't work when there are 2 n-n refs from the same 2 tables (but in different fields). Should instead create a global set of used table names.